syntax = "proto3";

package pb;

option go_package = "github.com/moverq1337/VTBHack/internal/pb";  

service NLPService {
  rpc ParseResume(ParseRequest) returns (ParseResponse);
  rpc MatchResumeVacancy(MatchRequest) returns (MatchResponse);  // Новый метод
}

message ParseRequest {
  string text = 1;
}

message ParseResponse {
  string parsed_data = 1;  // JSON с навыками, опытом и т.д.
}

message MatchRequest {
  string resume_text = 1;
  string vacancy_text = 2;
}

message MatchResponse {
  float score = 1;  // Косинусное сходство (0-1)
}